favor chromedriver over phantomjs

mattwr18 opened the pull request:
fixes#16

mattwr18 committed:
Favor chromedriver over phantmojs

mattwr18 committed:
Configure to use chrome_headless

mattwr18 committed:
Remove unused url host from test config

FedericoEsparza reviewed (commented):
@mattwr18 Shall I merge this?

mattwr18 commented:
oh hey @FedericoEsparza sorry, I meant to reach out to you and see if you wouldn't mind merging some `PR` for us... I know you're busy, but if feels weird making changes to `master`

This is ready to be merged, I believe, we have left the c9 code just in case we want to go back to using phantomjs

FedericoEsparza reviewed (commented):
I don't know that I can merge this.

It had a lot of issues on my computer (running phx 1.4)

I pushed up a branch [PR-review-branch-16- ](https://github.com/AgileVentures/mango/compare/16-favor-chromedriver-over-phantomjs...PR-review-branch-16-?expand=1)

I made several error-driven updates in pushing to simply run `mix test` with chromedriver. (i.e. `chromedriver & mix test && pkill chromedriver`)

I didn't have enough time to get it to work.

I got as far as a BEAM (OTP compiled version) issue:
<img width="1279" alt="Screen Shot 2019-03-20 at 7 24 49 AM" src="https://user-images.githubusercontent.com/11988089/54684645-c814ae00-4b0c-11e9-8f48-24c2af8ae3e4.png">

You're welcome to pull down the branch and troubleshoot.

Then you can use the above link to put in a PR to the original PR.

OR if it's working on your machines, feel free to just merge the original PR

Sorry for not reviewing this sooner.

mattwr18 commented:
sorry for the late reply, this highlights the issue with getting it working in one environment and not properly testing it in others... really appreciate your attention to this @FedericoEsparza

aonomike committed:
Use chromedriver

- Update hound in deps to use https instead of ssh for github
- delete order.tests

mattwr18 commented:
it's working for @aonomike on Mac, we haven't tried it with Phoneix 1.4 because we are on 1.3.
sorry for the ignorance, but I'm a bit confused when you say you are "issues on my computer (running phx 1.4)", I thought this was project based not computer based :thinking: 

We can certainly merge this, if you prefer, don't want cause extra work for you, but we are also not in a huge rush, maybe we'll go forward in the book just next week. 

@aonomike will put in a `PR` to set up the postgres info as environment variables. :smile_cat:

FedericoEsparza committed:
fix postgres(s) values in test.exs

FedericoEsparza committed:
Add .elixir_ls to .gitignore

FedericoEsparza committed:
Add .tool-versions

FedericoEsparza committed:
Update bcrypt_elixir

FedericoEsparza committed:
Add .tool-versions with elixir 1.7-otp-20

FedericoEsparza committed:
Update to Elixir 1.7 in mix.exs

FedericoEsparza committed:
Update Bcrypt functions

mattwr18 committed:
Update README

Co-authored-by: Mike Aono <aonomike@gmail.com>

aonomike committed:
Add travis configuration

mattwr18 committed:
Update travis with postgres

mattwr18 committed:
Use trusty instead

- issue with jdk version preinstalled?

Co-authored-by: Mike Aono <aonomike@gmail.com>

mattwr18 committed:
Change postgres version to 10

- 11 not supported by trusty

Co-authored-by: Mike Aono <aonomike@gmail.com>

mattwr18 committed:
Remove postgres versioning, cp config

Co-authored-by: Mike Aono <aonomike@gmail.com>
